src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -1112,7 +1112,12 @@ template<typename Key, typename Value>
 model::record_batch as_record_batch(Key key, Value val) {
     storage::record_batch_builder rb{
       model::record_batch_type::raft_data, model::offset{0}};
-    rb.add_raw_kv(to_json_iobuf(std::move(key)), to_json_iobuf(std::move(val)));
+    if constexpr (std::is_same_v<Value, std::nullopt_t>) {
```

Comment:
this constexpr expression looks curious. like what if `Value` isn't the literal type `std::nullopt_t` but is still a std::optional that contains no value?

Reply:
Yeah that's a great point. I think `is_convertible` would be the right choice. Either way this is no longer needed.